pkg/controllernamespace/namespace/namespace_controller.go

```diff
@@ -0,0 +1,132 @@
+
+	"github.com/openshift/openshift-azure/pkg/common"
+)
+
+// Add creates a new CustomerAdminReconciler Controller and adds it to the Manager. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager) error {
+	return add(mgr, newReconciler(mgr))
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+	return &ReconcileNamespace{client: mgr.GetClient(), scheme: mgr.GetScheme()}
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New("customeradminreconciler-namespace-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// TODO: figure out how to filter on namespaces that are !prefixed by a certain string
+	err = c.Watch(&source.Kind{Type: &corev1.Namespace{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+var _ reconcile.Reconciler = &ReconcileNamespace{}
+
+// ReconcileNamespace reconciles a CustomerAdminReconciler object
+type ReconcileNamespace struct {
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	client client.Client
+	scheme *runtime.Scheme
+}
+
+// Reconcile receives a namespace event and reconciles rolebindings for
+// customer roles
+// Note:
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileNamespace) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	if !common.RestrictedNamespace(request.Name) {
+		// move to appropriate place
+		// If we are namespace we need to reconcile 3 rolebindings:
+		// - customer-admin
+		// - customer-project-admin
+		// - customer-view
+		logrus.Printf("Reconciling Namespace %s/%s", request.Namespace, request.Name)
+
+		// check for existence of namespace
+		ns := &corev1.Namespace{}
+		err := r.client.Get(context.TODO(), request.NamespacedName, ns)
+		if err != nil {
+			// namespace is not found
+			if kerrors.IsNotFound(err) {
+				logrus.Printf("error: %v", err)
+				return reconcile.Result{}, nil
+			}
+		}
+		// do not reconcile. namespace is being deleted
+		if ns.Status.Phase == corev1.NamespaceTerminating {
+			logrus.Printf("namespace is being deleted %v", request.Namespace)
+			return reconcile.Result{}, nil
+		}
+
+		nn := types.NamespacedName{
+			Namespace: request.Name,
+		}
+		for _, rbname := range common.RoleBindingNames {
+			nrb := common.CustomerRoleBindingMap()(rbname)
+			nrb.Namespace = request.Name
+			nn.Name = rbname
+			rb := &rbacv1.RoleBinding{}
+			err := r.client.Get(context.TODO(), nn, rb)
+			if err != nil {
```

Comment:
see comment below, but in brief: if it exists, `return reconcile.Result{}, nil`.  If it doesn't, try to create.  If your create succeded, or if it failed because the object already exists (lost the race) `return reconcile.Result{}, nil`.  Otherwise, `return reconcile.Result{}, err`.